gym-unity/README.md

```diff
@@ -175,3 +180,147 @@ def main():
+
+```python
+from gym_unity.envs import UnityEnv
+```
+
+to import the Gym Wrapper. Navigate to the `create_atari_environment` method 
+in the same file, and switch to instantiating a Unity environment by replacing
+the method with the following code. 
+
+```python
+    game_version = 'v0' if sticky_actions else 'v4'
+    full_game_name = '{}NoFrameskip-{}'.format(game_name, game_version)
+    env = UnityEnv('./envs/GridWorld', 0, use_visual=True, uint8_visual=True)
+    return env
+```
+
+`./envs/GridWorld` is the path to your built Unity executable. For more information on 
+building Unity environments, see [here](../docs/Learning-Environment-Executable.md), and note 
+the Limitations section below. 
+
+Note that we are not using the preprocessor from Dopamine, 
+as it uses many Atari-specific calls. Furthermore, frame-skipping can be done from within Unity, 
+rather than on the Python side. 
+
+### Limitations
+
+Since Dopamine is designed around variants of DQN, it is only compatible
+with discrete action spaces, and specifically the Discrete Gym space. For environments
+that use branched discrete action spaces (e.g. 
+[VisualBanana](../docs/Learning-Environment-Examples.md)), you can enable the 
+`flatten_branched` parameter in `UnityEnv`, which treats each combination of branched 
+actions as separate actions.
+
+Furthermore, when building your environments, ensure that your
+[Learning Brain](../docs/Learning-Environment-Design-Brains.md) is using visual
+observations with greyscale enabled, and that the dimensions of the visual observations
+is 84 by 84 (matches the parameter found in `dqn_agent.py` and `rainbow_agent.py`).
+Dopamine's agents currently do not automatically adapt to the observation 
+dimensions or number of channels.  
+
+### Hyperparameters
+
+The hyperparameters provided by Dopamine are tailored to the Atari games, and you will
+likely need to adjust them for ML-Agents environments. Here is a sample 
+`dopamine/agents/rainbow/configs/rainbow.gin` file that is known to work with 
+GridWorld. 
+
+```python
+import dopamine.agents.rainbow.rainbow_agent
+import dopamine.unity.run_experiment
+import dopamine.replay_memory.prioritized_replay_buffer
+import gin.tf.external_configurables
+
+RainbowAgent.num_atoms = 51
+RainbowAgent.stack_size = 1
+RainbowAgent.vmax = 10.
+RainbowAgent.gamma = 0.99
+RainbowAgent.update_horizon = 3
+RainbowAgent.min_replay_history = 20000  # agent steps
+RainbowAgent.update_period = 5
+RainbowAgent.target_update_period = 50  # agent steps
+RainbowAgent.epsilon_train = 0.1
+RainbowAgent.epsilon_eval = 0.01
+RainbowAgent.epsilon_decay_period = 50000  # agent steps
+RainbowAgent.replay_scheme = 'prioritized'
+RainbowAgent.tf_device = '/cpu:0'  # use '/cpu:*' for non-GPU version
+RainbowAgent.optimizer = @tf.train.AdamOptimizer()
+
+tf.train.AdamOptimizer.learning_rate = 0.00025
+tf.train.AdamOptimizer.epsilon = 0.0003125
+
+Runner.game_name = "Unity"
```

Comment:
What is this for? does it matter?

Reply:
No, but Dopamine's code will crash if no name is given.

Reply:
Added blurb about game_name